docs/resources/vpn_gateway.md

```diff
@@ -0,0 +1,196 @@
+  local_subnets      = ["192.168.0.0/24", "192.168.1.0/24"]
+  connect_subnet     = "192.168.2.0/24"
+  availability_zones = ["cn-north-4a", "cn-north-4b"]
+
+  master_eip {
+    id = var.eip_id1
+  }
+
+  slave_eip {
+    id = var.eip_id2
+  }
+}
+```
+
+### Creating a VPN gateway with EIPs
+
+```HCL
+variable "name" {}
+variable "vpc_id" {}
+variable "bandwidth_name1" {}
+variable "bandwidth_name2" {}
+
+resource "huaweicloud_vpn_gateway" "test" {
+  name               = var.name
+  vpc_id             = var.vpc_id
+  local_subnets      = ["192.168.0.0/24", "192.168.1.0/24"]
+  connect_subnet     = "192.168.2.0/24"
+  availability_zones = ["cn-north-4a", "cn-north-4b"]
+
+  master_eip {
+    bandwidth_name = var.bandwidth_name1
+    type           = "5_bgp"
+    bandwidth_size = 5
+    charge_mode    = "traffic"
+  }
+
+  slave_eip {
+    bandwidth_name = var.bandwidth_name2
+    type           = "5_bgp"
+    bandwidth_size = 5
+    charge_mode    = "traffic"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `name` - (Required, String) The name of the VPN gateway. Only letters, digits, underscores(_) and hypens(-) are supported.
+
+* `vpc_id` - (Required, String, ForceNew) The ID of the VPC to which the VPN gateway is connected.
+
+  Changing this parameter will create a new resource.
+
+* `local_subnets` - (Required, List) The list of local subnets.
+
+* `connect_subnet` - (Required, String, ForceNew) The VPC network segment used by the VPN gateway needs to select an
+  independent network segment in the VPC for the VPN gateway to use, and cannot overlap with the existing subnet of the VPC.
+
+  Changing this parameter will create a new resource.
+
+* `availability_zones` - (Required, List, ForceNew) The list of availability zone IDs.
+
+  Changing this parameter will create a new resource.
+
+* `master_eip` - (Required, String, ForceNew) The master EIP configurations.
+  The [object](#Gateway_CreateRequestEip) structure is documented below.
+
+  Changing this parameter will create a new resource.
+
+* `slave_eip` - (Required, String, ForceNew) The slave EIP configurations.
+  The [object](#Gateway_CreateRequestEip) structure is documented below.
+
+  Changing this parameter will create a new resource.
+
+* `attachment_type` - (Optional, String, ForceNew) The attachment type. The value can be **vpc**.
+  Defaults to **vpc**
+
+  Changing this parameter will create a new resource.
+
+* `flavor` - (Optional, String, ForceNew) The flavor of the VPN gateway. The value can be **V1G** and **V300**.
+  Defaults to **V300**
+
+  Changing this parameter will create a new resource.
+
+* `bgp_asn` - (Optional, Int, ForceNew) The ASN number of BGP. The value ranges from **1** to **4294967295**.
```

Comment:
as we discussed before, this parameter should be named `asn`?

Reply:
fixed